packages/wrangler/src/publish/list.ts

```diff
@@ -0,0 +1,35 @@
+import { fetchResult } from "../cfetch";
+import { logger } from "../logger";
+
+export type ServicesListRes = [
+	{
+		id: string;
+		created_on: string;
+		default_environment: {
+			script: {
+				last_deployed_from: string;
+			};
+		};
+	}
+];
+
+export async function list(
+	accountId: string | undefined,
+	name: string | undefined
+) {
+	let url = `/accounts/${accountId}/workers/services`;
+	if (name) {
+		url += `?name=${name}`;
+	}
+
+	const services = await fetchResult<ServicesListRes>(url);
+
+	const workersMsg = services.map(
+		(service) =>
+			`\n${service.id}
+Deployed from: ${service.default_environment.script.last_deployed_from}
+Created on:    ${service.created_on}\n`
+	);
+
+	logger.log(...workersMsg);
```

Comment:
Just an alternative way of doing this (change not needed)
```suggestion
for(service of services){
logger.log(
			`\n${service.id}
Deployed from: ${service.default_environment.script.last_deployed_from}
Created on:    ${service.created_on}\n`
	)
	}
```